NextcloudTalk/CallKitManager.m

```diff
@@ -235,15 +235,19 @@ - (void)getCallInfoForCall:(CallKitCall *)call
     NCRoom *room = [[NCRoomsManager sharedInstance] roomWithToken:call.token forAccountId:call.accountId];
     if (room) {
         [self updateCall:call withDisplayName:room.displayName];
-    } else {
-        TalkAccount *account = [[NCDatabaseManager sharedInstance] talkAccountForAccountId:call.accountId];
-        [[NCAPIController sharedInstance] getRoomForAccount:account withToken:call.token withCompletionBlock:^(NSDictionary *roomDict, NSError *error) {
-            if (!error) {
-                NCRoom *room = [NCRoom roomWithDictionary:roomDict andAccountId:call.accountId];
-                [self updateCall:call withDisplayName:room.displayName];
-            }
-        }];
     }
+    
+    TalkAccount *account = [[NCDatabaseManager sharedInstance] talkAccountForAccountId:call.accountId];
+    [[NCAPIController sharedInstance] getRoomForAccount:account withToken:call.token withCompletionBlock:^(NSDictionary *roomDict, NSError *error) {
+        if (!error) {
+            NCRoom *room = [NCRoom roomWithDictionary:roomDict andAccountId:call.accountId];
+            [self updateCall:call withDisplayName:room.displayName];
+            
+            NSInteger callFlag = [[roomDict objectForKey:@"callFlag"] integerValue];
```

Comment:
No capability-check needed here?

Reply:
No. If the "callFlag" attribute is not there, callFlag will be 0. So it will notify an audio-only call as in the past when no "callFlag" attribute was sent.

Reply:
Okay, wasn't sure if you would get an NSInteger for a nil value.

Reply:
No problem ;) Thanks a lot for taking a look at the code :)
But yeah, [nil integeValue] will return 0.

Reply:
Lazy. 0 could mean everyone left already again. We added a capability for this so you could trust the value:
https://nextcloud-talk.readthedocs.io/en/latest/capabilities/#111

But on iOS you have to present a call screen anyway, right? Even if the other person already left again

Reply:
> But on iOS you have to present a call screen anyway, right? Even if the other person already left again

Yes. That's why I can only differentiate between presenting and audio or a video incoming call :D

Reply:
But we could add this in https://github.com/nextcloud/talk-ios/pull/492 to end the call when the other person already left.

Reply:
Hm, but you could end the call immediatly if the capability is there and it's 0, right? At this point the call was already reported and we can hang up again.

Reply:
You are right! We could add it in a following PR ;)

Reply:
Implemented in #507